src/globus_sdk/services/transfer/data/transfer_data.py

```diff
@@ -184,6 +188,7 @@ def __init__(
         # garbage overnight
         if sync_level is not None:
             sync_dict = {"exists": 0, "size": 1, "mtime": 2, "checksum": 3}
+            # TODO: sync_level not allowed to be int?
```

Comment:
I think `sync_level: Union[None, str, int]` would be best. We use the `get()` call below to handle the `int` case, and we should allow the int since that's what the API and API docs use.